core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/reindex/ReindexRequestBuilder.java

```diff
@@ -41,4 +42,9 @@ public ReindexRequestBuilder setFailedCommitTimestamp(long failedCommitTimestamp
 		return req;
 	}
 
+	@Override
+	public Health[] allowedHealthstates() {
+		// allowing reindexing requests to execute in all health states.
+		return Health.values(); 
```

Comment:
`Health.RED` should indicate major problems with the repository. Reject all requests sent to repositories in this state.